crates/cext/src/transpiler/target.rs

```diff
@@ -869,6 +872,474 @@ pub unsafe extern "C" fn qk_target_num_instructions(target: *const Target) -> us
     target.len()
 }
 
+/// @ingroup QkTarget
+/// Return the index in which an operation is located based on its name.
+///
+/// @param target A pointer to the ``QkTarget``.
+/// @param name A pointer to the name string which we will look up an index for.
+///
+/// # Example
+/// ```c
+///     QkTarget *target = qk_target_new(5);
+///     QkTargetEntry *target_entry = qk_target_entry_new(QkGate_H);
+///     qk_target_add_instruction(target, target_entry);
```

Comment:
This might invalidate the particular example, but can we also add an "out" parameter to `qk_target_add_instruction` so that we have the index without needing to do a costly lookup, e.g.:

```rust
pub unsafe extern "C" fn qk_target_add_instruction(
    target: *mut Target,
    target_entry: *mut TargetEntry,
    out_index: *mut usize,
) -> ExitCode { ... }
```

You can document that the user can pass `out_index` as `NULL` if they don't need it.